packages/grafana-ui/src/components/GraphNG/GraphNG.tsx

```diff
@@ -20,24 +20,30 @@ import { LegendDisplayMode, LegendItem, LegendOptions } from '../Legend/Legend';
 import { GraphLegend } from '../Graph/GraphLegend';
 import { UPlotConfigBuilder } from '../uPlot/config/UPlotConfigBuilder';
 import { useRevision } from '../uPlot/hooks';
+import { GraphNGLegendEvent, GraphNGLegendEventMode } from './types';
 
 const defaultFormatter = (v: any) => (v == null ? '-' : v.toFixed(1));
 
 export interface XYFieldMatchers {
   x: FieldMatcher;
   y: FieldMatcher;
 }
-
 export interface GraphNGProps extends Omit<PlotProps, 'data' | 'config'> {
   data: DataFrame[];
   legend?: LegendOptions;
   fields?: XYFieldMatchers; // default will assume timeseries data
+  onLegendClick?: (event: GraphNGLegendEvent) => void;
 }
 
 const defaultConfig: GraphFieldConfig = {
   drawStyle: DrawStyle.Line,
   showPoints: PointVisibility.Auto,
   axisPlacement: AxisPlacement.Auto,
+  seriesConfig: {
```

Comment:
maybe `fieldVisibility`?  or `showFieldIn` or `showField`

avoid "DisplayField" since that is too close to [FieldDisplay](https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/field/fieldDisplay.ts#L65)

Reply:
Since we want the default to be visible in everything, could we have the options be:
```
hideFrom: {
   graph: false,
   legend: false,
   tooltip: false,
}
```

That will make the UI more complicated, but then checking for `config.hideFrom?.graph` would default more smoothly

Reply:
Great suggestions!

Reply:
@ryantxu yups, we discussed this with @mckn yesterday and it feels like showIn* option in the UI is super confusing, especially if you look at the overrides config.